game-core/src/main/java/games/strategy/engine/data/GameMap.java

```diff
@@ -103,15 +104,24 @@ public Territory getTerritory(final String s) {
     return neighbors;
   }
 
+  public Set<Territory> getNeighbors(
+      final Territory territory, @Nullable final Predicate<Territory> cond) {
+    return getNeighbors(territory, cond, (it, it2) -> true);
+  }
+
   /**
    * Returns all adjacent neighbors of the starting territory that match the condition. Does NOT
    * include the original/starting territory in the returned Set.
    */
-  public Set<Territory> getNeighbors(final Territory territory, final Predicate<Territory> cond) {
+  public Set<Territory> getNeighbors(
+      final Territory territory,
+      @Nullable final Predicate<Territory> cond,
```

Comment:
Nit - This parameter is now the 'territoryCondition', correct? Would you have any objection to rename it?

Now that there are two parameters, making the names distinct is more significant as one wonders how the 'cond' and 'routeCond' are different. Secondly, 'cond' was not the best name to begin with as it was cryptic and did not describe what the condition represented or was used for.

Reply:
It looks like this method is only called internally to `GameMap` now as well, if I have that right - we can reduce the visibility to `private`. Would you mind doing so?